src/kml/src/main/java/org/geoserver/kml/builder/SimpleNetworkLinkBuilder.java

```diff
@@ -68,8 +73,20 @@ void encodeDocumentContents(Document container) {
         for (int i = 0; i < layers.size(); i++) {
             MapLayerInfo layerInfo = layers.get(i);
             NetworkLink nl = container.createAndAddNetworkLink();
-            nl.setName(layerInfo.getName());
-            nl.setVisibility(true);
+            nl.setName(layerInfo.getLabel());
```

Comment:
This line is probably the change that created the failure (based on cite:BasicPolygons vs BasicPolygons)